src/components/base/BaseInput.vue

```diff
@@ -1,35 +1,67 @@
 <script setup lang="ts">
+import { ref, computed } from "vue";
+import EyeIcon from "@/assets/icons/eye.svg?component";
+
 const props = withDefaults(
   defineProps<{
     modelValue?: string;
     inputType?: string;
     inputName?: string;
     placeholder?: string;
+    validation?: string;
   }>(),
   {
     modelValue: "",
     inputValue: "text",
     inputName: "",
     placeholder: "...",
+    validation: "",
   }
 );
+
+const passwordVisible = ref(false);
+
+const toggleViewPassword = () => {
+  passwordVisible.value = !passwordVisible.value;
+};
+
+const inputClasses = computed(() => {
+  return [
+    "base-input__input",
+    props.validation ? "base-input__input--error" : "",
+  ];
+});
 </script>
 
 <template>
   <div class="base-input">
-    <label :for="props.inputName" class="base-input__label"><slot /></label>
-    <input
-      :id="props.inputName"
-      class="base-input__input"
-      required
-      :type="props.inputType"
-      :placeholder="props.placeholder"
-      :modelValue="props.modelValue"
-      :name="props.inputType"
-      @input="
-        $emit('update:modelValue', ($event.target as HTMLInputElement).value)
-      "
-    />
+    <label :for="props.inputName" class="base-input__label"
+      ><slot />
+      <input
+        :id="props.inputName"
+        :class="inputClasses"
+        required
+        :type="passwordVisible ? 'text' : inputType"
+        :placeholder="props.placeholder"
+        :modelValue="props.modelValue"
+        :name="props.inputType"
+        @input="
+          $emit('update:modelValue', ($event.target as HTMLInputElement).value)
+        "
+        @blur="$emit('blur')"
+      />
+      <button
+        v-if="props.inputType === 'password'"
+        type="button"
+        class="base-input__button"
+        @click="toggleViewPassword"
+      >
+        <EyeIcon />
+      </button>
+    </label>
+    <div v-if="props.validation" class="base-input__validations">
```

Comment:
you don't need to call `props.validation` while you're in here just `validation` should work fine

Reply:
I found about this well into the project - jsut decided to keep the convention on files I created eralier - like Base button or input